nixos/modules/programs/spacefm.nix

```diff
@@ -0,0 +1,41 @@
+# Global configuration for spacefm.
+
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.programs.spacefm;
+
+in
+{
+  ###### interface
+
+  options = {
+
+    programs.spacefm = {
+
+      settings = mkOption {
+        type = types.attrs;
+        default = {};
+        example = literalExample ''{
```

Comment:
yeah I already figured out an easier way by simply importing the _spacefm.nix_ in _configuration.nix._

I've also managed to test and show I had nothing to be concerned about. I add pkgs.zile to a the default as a test value and it didn't try to pull it until I enabled the module. So, it won't evaluate the default so long as it's disabled.

I'll update my copy and push now...